smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/providers/HttpQueryItemProvider.kt

```diff
@@ -70,10 +70,18 @@ class HttpQueryItemProvider(
     }
 
     fun renderProvider(writer: SwiftWriter) {
-        writer.openBlock("extension \$N: \$N {", "}", inputSymbol, ClientRuntimeTypes.Middleware.Providers.QueryItemProvider) {
-            writer.openBlock("public var queryItems: [\$N] {", "}", ClientRuntimeTypes.Core.URLQueryItem) {
-                writer.openBlock("get throws {", "}") {
-                    writer.write("var items = [\$N]()", ClientRuntimeTypes.Core.URLQueryItem)
+        writer.openBlock("extension \$N {", "}", inputSymbol) {
+            writer.write("")
+            writer.openBlock(
+                "static func queryItemProvider(_ value: \$N) throws -> [\$N] {",
+                "}",
+                inputSymbol,
+                ClientRuntimeTypes.Core.SDKURLQueryItem
+            ) {
+                if (queryLiterals.isEmpty() && queryBindings.isEmpty()) {
+                    writer.write("return []")
+                } else {
+                    writer.write("var items = [\$N]()", ClientRuntimeTypes.Core.SDKURLQueryItem)
```

Comment:
- Removed the protocol from the extension
- Changed property to a static function that is of type `QueryItemProvider<inputSymbol>`
- If there are no query-bound items, return a literal `[]` to avoid a compiler warning for unmodified `var`